### PR TITLE
Fix Python 3 syntax and test failures

### PR DESCRIPTION
**Pull request by kahowell**
After this lands, we will be able to run tests via Fedora with
nosetests-3. Additionally the bin scripts should also work with Python
3. For example, `python3 bin/subscription-manager` should work (with the
proper `PYTHONPATH`).

### CODE REVIEW

File: test/test_utils.py [hunk 1]
```diff
@@ -536,7 +536,6 @@ def test_join(self):
         self.assertEqual("One", friendly_join(["One"]))
         self.assertEqual("One and Two", friendly_join(["One", "Two"]))
         self.assertEqual("One, Two, and Three", friendly_join(["One", "Two", "Three"]))
-        self.assertEqual("Three, Two, and One", friendly_join(set(["One", "Two", "Three"])))
```
Comment (jirihnidek) [inline test/test_utils.py]: Interesting difference between Python 2 and Python 3.

File: test/test_rct_manifest_command.py [hunk 2]
```diff
@@ -41,10 +41,10 @@
 
 
 def _build_valid_manifest():
-    manifest_zip = six.StringIO()
+    manifest_zip = six.BytesIO()
```
Comment (jirihnidek) [inline test/test_rct_manifest_command.py]: I'm just curious. Why did you replace `StringIO` with `BytesIO`?

File: test/test_productid.py [hunk 3]
```diff
@@ -9,7 +9,6 @@
 import os
 import shutil
 import tempfile
-import yum
```
Comment (jirihnidek) [inline test/test_productid.py]: Is removing of yum in this test related to removing Alex's effort of removing dependency to yum?

File: test/test_managercli.py [hunk 4]
```diff
@@ -1560,13 +1567,14 @@ def test_he_ssl_error(self):
...
-        except SystemExit as e:
-            self.assertEqual(e.code, os.EX_SOFTWARE)
+        if six.PY2:
+            e = SSL.Checker.WrongHost("expectedHost.example.com",
+                                       "actualHost.example.com",
+                                       "subjectAltName")
+            try:
+                managercli.handle_exception("huh", e)
+            except SystemExit as e:
+                self.assertEqual(e.code, os.EX_SOFTWARE)
```
Comment (jirihnidek) [inline test/test_managercli.py]: What about Python 3?

File: test/test_lock.py [hunk 5]
```diff
@@ -73,7 +73,7 @@ def wait_for_pid(timer):
 
     def close_lock_holder(self):
         try:
-            self.other_process.communicate("whatever")
+            self.other_process.communicate("whatever".encode('utf-8'))
```
Comment (jirihnidek) [inline test/test_lock.py]: BTW: this: `cd src; grep -R -i 'encode("utf' * | wc -l` gives me 2 :-).

File: test/test_about.py [hunk 6]
```diff
@@ -21,18 +21,24 @@
...
+import six
 
 from subscription_manager.ga import Gtk as ga_Gtk
 from subscription_manager.gui import about
 
+if six.PY2:
+    OPEN_METHOD = '__builtin__.open'
+else:
+    OPEN_METHOD = 'builtins.open'
+
```
[the remark this answers is not in the document]
Comment (kahowell) [reply test/test_about.py]: Small oversight. Thanks for catching that.

File: test/test_managercli.py (hunk 4 above)
Comment (kahowell) [reply test/test_managercli.py]: This is a test of a M2Crypto-specific interface. I'll change it to `raise SkipTest` to be more clear.

File: test/test_rct_manifest_command.py (hunk 2 above)
Comment (kahowell) [reply test/test_rct_manifest_command.py]: It was necessary to get the test to pass. If you consider that a zip file is handled as binary data, it makes sense.

File: test/test_lock.py (hunk 5 above)
Comment (kahowell) [reply test/test_lock.py]: Changing this and other instances where it makes sense to unicode literals (`u''` syntax)...

Comment (kahowell) [reply test/test_lock.py]: Er... nevermind. That doesn't really make sense and it broke things. Coffee hasn't kicked in yet. :coffee:

File: test/test_productid.py (hunk 3 above)
Comment (kahowell) [reply test/test_productid.py]: No, this is just dead code removal, resulting in a no-longer-needed import 😄

**Comment by jirihnidek** [comment]
Just one note: I had to install all following rpm packages on Fedora 26 to be able to avoid some failures in `test_po_files.py`, because Fedora does install only requested language packages.

```console
dnf install langpacks-bn langpacks-es langpacks-fr langpacks-gu langpacks-hi langpacks-it langpacks-kn langpacks-ml langpacks-mr langpacks-pa langpacks-pt langpacks-ru langpacks-ta langpacks-te langpacks-zh_CN langpacks-zh_TW langpacks-ko langpacks-or
```

**Comment by kahowell** [comment]
@jirihnidek applied the fix you suggested, and then fixed a few more (more usages of `__builtin__`). Must have missed them in my first few passes...

**Comment by kahowell** [comment]
e4ee298a..5fd28466: squashed

**Pull request merged**